nifi-api/src/main/java/org/apache/nifi/provenance/ProvenanceEventBuilder.java

```diff
@@ -283,6 +283,13 @@ public interface ProvenanceEventBuilder {
      */
     ProvenanceEventBuilder setRelationship(Relationship relationship);
 
+    /**
+     * Sets the IDs for the event that happened previously to this event for the given FlowFile
+     * @param previousEventIds The previous event IDs (usually one except for JOIN events and such)
+     * @return the builder
+     */
+    ProvenanceEventBuilder setPreviousEventIds(List<Long> previousEventIds);
```

Comment:
Does this imply we're storing event ids for the before and after event on both the before and after event?

Reply:
Ah maybe it is that we had parent FlowFile identifiers and now we'll have previous Event Identfiers.